types/src/error.rs

```diff
@@ -19,12 +20,22 @@ impl<T: fmt::Display> fmt::Display for Mismatch<T> {
 /// Error that occurs when a call failed.
 #[derive(Debug, thiserror::Error)]
 pub enum CallError {
```

Comment:
we might want to put `non_exhaustive` on this but a little weird to just add some error code for non defined enum variant :P